test/test.sh

```diff
@@ -85,6 +102,16 @@ tmpfile=$(mktemp)
 test_expected_files "${tmpfile}" "${test_name}"
 test_expected_keys "${tmpfile}" "${test_name}"
 
+# See https://github.com/ashcrow/filetranspiler/pull/29
+test_name="Ignition Overwrite With Fakeroot File"
+tmpfile=$(mktemp)
+./filetranspile -i test/ignition-overwrite-with-fakeroot-file.json -f test/fakeroot > ${tmpfile}
+test_expected_files "${tmpfile}" "${test_name}"
+test_expected_keys "${tmpfile}" "${test_name}"
+# We should still have the fakeroot one and not the one in our original ignition
+test_expected_contents "${tmpfile}" "/etc/resolve.conf" "data:text/plain;charset=us-ascii;base64,c2VhcmNoIDEyNy4wLjAuMQpuYW1lc2VydmVyIDEyNy4wLjAuMQo=" "${test_name}"
```

Comment:
```suggestion
test_expected_contents "${tmpfile}" "/etc/resolve.conf" "data:text/plain;base64,c2VhcmNoIDEyNy4wLjAuMQpuYW1lc2VydmVyIDEyNy4wLjAuMQo=" "${test_name}"
```

Tests fails with :
```
PASS: Ignition Overwrite With Fakeroot File: File Check
PASS: Ignition Overwrite With Fakeroot File: Key Check
FAIL: Ignition Overwrite With Fakeroot File: Content Check for /etc/resolve.conf Content does not match
- Expected: data:text/plain;charset=us-ascii;base64,c2VhcmNoIDEyNy4wLjAuMQpuYW1lc2VydmVyIDEyNy4wLjAuMQo=
- Got: data:text/plain;base64,c2VhcmNoIDEyNy4wLjAuMQpuYW1lc2VydmVyIDEyNy4wLjAuMQo=
```

Reply:
Interesting ... here is my output as is:

```
$ ./test/test.sh 
<snip/>
PASS: Ignition Overwrite With Fakeroot File: File Check
PASS: Ignition Overwrite With Fakeroot File: Key Check
PASS: Ignition Overwrite With Fakeroot File: Content Check for /etc/resolve.conf
```

And with the above change:

```
$ ./test/test.sh 
<snip/>
PASS: Ignition Overwrite With Fakeroot File: File Check
PASS: Ignition Overwrite With Fakeroot File: Key Check
FAIL: Ignition Overwrite With Fakeroot File: Content Check for /etc/resolve.conf Content does not match
- Expected: data:text/plain;base64,c2VhcmNoIDEyNy4wLjAuMQpuYW1lc2VydmVyIDEyNy4wLjAuMQo=
- Got: data:text/plain;charset=us-ascii;base64,c2VhcmNoIDEyNy4wLjAuMQpuYW1lc2VydmVyIDEyNy4wLjAuMQo=
```
Is it possible you have another change in your `filetranspiler` or run the test off an older version?

Reply:
I think it's sth related to magic library version:
```
#!/usr/bin/env python3
import magic

mimetype = magic.open(magic.MAGIC_MIME)
mimetype.load()
print(mimetype.file("./test/ignition-overwrite-with-fakeroot-file.json"))
```
What's the outputs of the above script for you? I only get the following output:
```
text/plain
```

So I think filetranspiler is behaving differently on some environments. What's the version of magic module for you?

One thing that might help, is if we had a `requirements.yml` file pining the version of modules which this have been tested with.

Reply:
Good point. 

`python3-file-magic-5.39-3.fc33.noarch`

Running the script I get:

```
application/json; charset=us-ascii
```

`requirements.yml` is a good idea _but_ it looks like different distributions may have different patches expected. I don't want to _force_ folks to use the container instead of the script directly, but it would take care of those differences. For the time being what about me updating the test to ignore ensure there is a file type based on the output of the `file -b --mime-encoding --mime-type $FILENAME`. Thoughts?

Reply:
Updated the PR with the above idea.

Reply:
Actually `file -b --mime-encoding --mime-type $FILENAME`  return this for me:
```
text/plain;charset=us-ascii
```
however, python-magic still returning `text/plain`, so the tests still fails and it doesn't help.

I solved this problem with changing filetransipler code check this commit in PR30: https://github.com/ashcrow/filetranspiler/pull/29/commits/0febf646b12a9083bcd6b3155a918b237a2d3d77

This way is even better as files generated on all environments will have same output.

I think it's ok if you want to revert this last change on test scripts to make it simpler. Now the tests work either with or without `file -b --mime-encoding --mime-type $FILENAME`.

Reply:
Awesome! Looking now.

Reply:
Since the tests work either way I'll leave it as is. Merging them both now. Thank you!!